src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.specialized.BlobLeaseAsyncClient;
+import com.azure.storage.blob.specialized.BlobLeaseClientBuilder;
+import org.slf4j.Logger;
+
+import java.util.List;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class ContainerProcessor {
+
+    private static final Logger LOGGER = getLogger(ContainerProcessor.class);
+
+    private static final BlobLeaseClientBuilder LEASE_CLIENT_BUILDER = new BlobLeaseClientBuilder();
```

Comment:
Should be. Building a client (calling method `buildAsyncClient`) creates a new client every time. Blob clients are bounded per blob item so have to build new client every time as well.

Reply:
I think it's not, based on the nature of builder pattern.
It's mutable and calling it may contain 'remains' from previous call.

Like here:
```java
var abc = BUILDER.setA("abc").setB(100).build()
...
var xyz = BUILDER.setB(100).build();
```
`xyz` now has also property `A` set to `"abc"`.

Please init builder for every usage.